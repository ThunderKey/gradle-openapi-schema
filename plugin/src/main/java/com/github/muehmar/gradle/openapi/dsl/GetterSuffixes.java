package com.github.muehmar.gradle.openapi.dsl;

import io.github.muehmar.pojobuilder.annotations.Nullable;
import io.github.muehmar.pojobuilder.annotations.PojoBuilder;
import java.io.Serializable;
import java.util.Optional;
import lombok.Data;
import lombok.With;

@Data
@With
@PojoBuilder
public class GetterSuffixes implements Serializable {
  private static final String DEFAULT_REQUIRED = "";
  private static final String DEFAULT_OPTIONAL = "Opt";
  private static final String DEFAULT_REQUIRED_NULLABLE = "Opt";
  private static final String DEFAULT_OPTIONAL_NULLABLE = "Tristate";

  @Nullable private String requiredSuffix;
  @Nullable private String requiredNullableSuffix;
  @Nullable private String optionalSuffix;
  @Nullable private String optionalNullableSuffix;

  public GetterSuffixes() {
    this(null, null, null, null);
  }

  public GetterSuffixes(
      String requiredSuffix,
      String requiredNullableSuffix,
      String optionalSuffix,
      String optionalNullableSuffix) {
    this.requiredSuffix = requiredSuffix;
    this.requiredNullableSuffix = requiredNullableSuffix;
    this.optionalSuffix = optionalSuffix;
    this.optionalNullableSuffix = optionalNullableSuffix;
  }

  public static GetterSuffixes allUndefined() {
    return new GetterSuffixes(null, null, null, null);
  }

  public GetterSuffixes withCommonSuffixes(GetterSuffixes commonSuffixes) {
    return new GetterSuffixes(
        getRequiredSuffix().orElse(commonSuffixes.getRequiredSuffix().orElse(null)),
        getRequiredNullableSuffix().orElse(commonSuffixes.getRequiredNullableSuffix().orElse(null)),
        getOptionalSuffix().orElse(commonSuffixes.getOptionalSuffix().orElse(null)),
        getOptionalNullableSuffix()
            .orElse(commonSuffixes.getOptionalNullableSuffix().orElse(null)));
  }

  public Optional<String> getRequiredSuffix() {
    return Optional.ofNullable(requiredSuffix);
  }

  public String getRequiredSuffixOrDefault() {
    return getRequiredSuffix().orElse(DEFAULT_REQUIRED);
  }

  public Optional<String> getRequiredNullableSuffix() {
    return Optional.ofNullable(requiredNullableSuffix);
  }

  public String getRequiredNullableSuffixOrDefault() {
    return getRequiredNullableSuffix().orElse(DEFAULT_REQUIRED_NULLABLE);
  }

  public Optional<String> getOptionalSuffix() {
    return Optional.ofNullable(optionalSuffix);
  }

  public String getOptionalSuffixOrDefault() {
    return getOptionalSuffix().orElse(DEFAULT_OPTIONAL);
  }

  public Optional<String> getOptionalNullableSuffix() {
    return Optional.ofNullable(optionalNullableSuffix);
  }

  public String getOptionalNullableSuffixOrDefault() {
    return getOptionalNullableSuffix().orElse(DEFAULT_OPTIONAL_NULLABLE);
  }
}
