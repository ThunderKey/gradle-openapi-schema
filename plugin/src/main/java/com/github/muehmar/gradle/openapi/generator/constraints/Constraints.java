package com.github.muehmar.gradle.openapi.generator.constraints;

import java.util.Objects;
import java.util.Optional;
import java.util.function.Consumer;

public class Constraints {
  private final Min min;
  private final Max max;
  private final DecimalMin decimalMin;
  private final DecimalMax decimalMax;
  private final Size size;
  private final Pattern pattern;
  private final Email email;

  private Constraints(
      Min min,
      Max max,
      DecimalMin decimalMin,
      DecimalMax decimalMax,
      Size size,
      Pattern pattern,
      Email email) {
    this.min = min;
    this.max = max;
    this.decimalMin = decimalMin;
    this.decimalMax = decimalMax;
    this.size = size;
    this.pattern = pattern;
    this.email = email;
  }

  public static Constraints empty() {
    return new Constraints(null, null, null, null, null, null, null);
  }

  public static Constraints ofMin(Min min) {
    return Constraints.empty().withMin(min);
  }

  public static Constraints ofMax(Max max) {
    return Constraints.empty().withMax(max);
  }

  public static Constraints ofMinAndMax(Min min, Max max) {
    return Constraints.empty().withMin(min).withMax(max);
  }

  public static Constraints ofSize(Size size) {
    return Constraints.empty().withSize(size);
  }

  public static Constraints ofPattern(Pattern pattern) {
    return Constraints.empty().withPattern(pattern);
  }

  public static Constraints ofEmail() {
    return Constraints.empty().withEmail(new Email());
  }

  public static Constraints ofDecimalMin(DecimalMin decimalMin) {
    return Constraints.empty().withDecimalMin(decimalMin);
  }

  public static Constraints ofDecimalMax(DecimalMax decimalMax) {
    return Constraints.empty().withDecimalMax(decimalMax);
  }

  public Constraints withMin(Min min) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withMax(Max max) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withSize(Size size) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withPattern(Pattern pattern) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withEmail(Email email) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withDecimalMin(DecimalMin decimalMin) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public Constraints withDecimalMax(DecimalMax decimalMax) {
    return new Constraints(min, max, decimalMin, decimalMax, size, pattern, email);
  }

  public void onMin(Consumer<Min> onMin) {
    Optional.ofNullable(min).ifPresent(onMin);
  }

  public void onMax(Consumer<Max> onMax) {
    Optional.ofNullable(max).ifPresent(onMax);
  }

  public void onDecimalMin(Consumer<DecimalMin> onDecimalMin) {
    Optional.ofNullable(decimalMin).ifPresent(onDecimalMin);
  }

  public void onDecimalMax(Consumer<DecimalMax> onDecimalMax) {
    Optional.ofNullable(decimalMax).ifPresent(onDecimalMax);
  }

  public void onSize(Consumer<Size> onSize) {
    Optional.ofNullable(size).ifPresent(onSize);
  }

  public void onPattern(Consumer<Pattern> onPattern) {
    Optional.ofNullable(pattern).ifPresent(onPattern);
  }

  public void onEmail(Consumer<Email> onEmail) {
    Optional.ofNullable(email).ifPresent(onEmail);
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Constraints that = (Constraints) o;
    return Objects.equals(min, that.min)
        && Objects.equals(max, that.max)
        && Objects.equals(size, that.size)
        && Objects.equals(pattern, that.pattern)
        && Objects.equals(email, that.email);
  }

  @Override
  public int hashCode() {
    return Objects.hash(min, max, size, pattern, email);
  }

  @Override
  public String toString() {
    return "Constraints{"
        + "min="
        + min
        + ", max="
        + max
        + ", size="
        + size
        + ", pattern="
        + pattern
        + ", email="
        + email
        + '}';
  }
}
