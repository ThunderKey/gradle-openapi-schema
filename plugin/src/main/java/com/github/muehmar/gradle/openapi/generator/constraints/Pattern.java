package com.github.muehmar.gradle.openapi.generator.constraints;

import java.util.Objects;

public class Pattern {
  private final String value;

  public Pattern(String value) {
    this.value = value;
  }

  public String getPattern() {
    return value;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Pattern pattern = (Pattern) o;
    return Objects.equals(value, pattern.value);
  }

  @Override
  public int hashCode() {
    return Objects.hash(value);
  }

  @Override
  public String toString() {
    return "Pattern{" + "value='" + value + '\'' + '}';
  }
}
