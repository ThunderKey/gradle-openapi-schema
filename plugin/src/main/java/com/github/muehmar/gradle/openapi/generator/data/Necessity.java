package com.github.muehmar.gradle.openapi.generator.data;

/** Defines the necessity of an attribute, i.e. if the attribute is required or optional. */
public enum Necessity {
  REQUIRED,
  OPTIONAL;

  public static Necessity fromBoolean(boolean necessity) {
    return necessity ? REQUIRED : OPTIONAL;
  }
}
