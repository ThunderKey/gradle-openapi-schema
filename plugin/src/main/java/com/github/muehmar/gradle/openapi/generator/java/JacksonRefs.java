package com.github.muehmar.gradle.openapi.generator.java;

public class JacksonRefs {
  private JacksonRefs() {}

  public static final String JSON_POJO_BUILDER =
      "com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder";
}
