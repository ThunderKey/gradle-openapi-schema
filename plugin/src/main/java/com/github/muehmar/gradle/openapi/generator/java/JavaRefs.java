package com.github.muehmar.gradle.openapi.generator.java;

public class JavaRefs {
  private JavaRefs() {}

  public static final String LOCAL_DATE = "java.time.LocalDate";
  public static final String OPTIONAL = "java.util.Optional";
}
