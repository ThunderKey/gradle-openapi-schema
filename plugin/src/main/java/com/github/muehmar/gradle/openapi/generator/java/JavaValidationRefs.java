package com.github.muehmar.gradle.openapi.generator.java;

public class JavaValidationRefs {
  private JavaValidationRefs() {}

  public static final String NOT_NULL = "javax.validation.constraints.NotNull";
  public static final String PATTERN = "javax.validation.constraints.Pattern";
  public static final String ASSERT_TRUE = "javax.validation.constraints.AssertTrue";
}
