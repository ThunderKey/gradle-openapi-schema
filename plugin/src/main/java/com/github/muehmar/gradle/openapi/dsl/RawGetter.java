package com.github.muehmar.gradle.openapi.dsl;

import static com.github.muehmar.gradle.openapi.util.Optionals.or;

import ch.bluecare.commons.data.PList;
import com.github.muehmar.gradle.openapi.generator.settings.JavaModifier;
import io.github.muehmar.pojoextension.annotations.Nullable;
import io.github.muehmar.pojoextension.annotations.PojoExtension;
import java.io.Serializable;
import java.util.Optional;
import lombok.Data;
import org.gradle.api.InvalidUserDataException;

@Data
@PojoExtension
public class RawGetter implements Serializable, RawGetterExtension {
  private static final String DEFAULT_MODIFIER = JavaModifier.PRIVATE.getValue();
  private static final String DEFAULT_SUFFIX = "Raw";
  private static final boolean DEFAULT_DEPRECATED_ANNOTATION = false;

  @Nullable String modifier;
  @Nullable String suffix;
  @Nullable Boolean deprecatedAnnotation;

  public RawGetter() {
    this(null, null, null);
  }

  public RawGetter(String modifier, String suffix, Boolean deprecatedAnnotation) {
    this.modifier = modifier;
    this.suffix = suffix;
    this.deprecatedAnnotation = deprecatedAnnotation;
  }

  public static RawGetter allUndefined() {
    return new RawGetter();
  }

  public RawGetter withCommonRawGetter(RawGetter commonRawGetter) {
    return RawGetterBuilder.create()
        .andAllOptionals()
        .modifier(or(getModifier(), commonRawGetter.getModifier()))
        .suffix(or(getSuffix(), commonRawGetter.getSuffix()))
        .deprecatedAnnotation(
            or(getDeprecatedAnnotation(), commonRawGetter.getDeprecatedAnnotation()))
        .build();
  }

  public Optional<String> getModifier() {
    return Optional.ofNullable(modifier);
  }

  public JavaModifier getModifierOrDefault() {
    final String modifierString = getModifier().orElse(DEFAULT_MODIFIER);

    return JavaModifier.fromString(modifierString)
        .orElseThrow(
            () ->
                new InvalidUserDataException(
                    "Unsupported value for modifier: '"
                        + modifierString
                        + "'. Supported values are ["
                        + PList.of(JavaModifier.values())
                            .map(JavaModifier::getValue)
                            .mkString(", ")));
  }

  public Optional<String> getSuffix() {
    return Optional.ofNullable(suffix);
  }

  public String getSuffixOrDefault() {
    return getSuffixOr(DEFAULT_SUFFIX);
  }

  public Optional<Boolean> getDeprecatedAnnotation() {
    return Optional.ofNullable(deprecatedAnnotation);
  }

  public boolean getDeprecatedAnnotationOrDefault() {
    return getDeprecatedAnnotationOr(DEFAULT_DEPRECATED_ANNOTATION);
  }
}
