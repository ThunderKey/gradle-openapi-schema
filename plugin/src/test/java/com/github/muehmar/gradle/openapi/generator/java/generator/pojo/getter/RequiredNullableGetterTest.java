package com.github.muehmar.gradle.openapi.generator.java.generator.pojo.getter;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import com.github.muehmar.gradle.openapi.generator.java.JacksonRefs;
import com.github.muehmar.gradle.openapi.generator.java.JavaRefs;
import com.github.muehmar.gradle.openapi.generator.java.JavaValidationRefs;
import com.github.muehmar.gradle.openapi.generator.java.model.JavaPojoMember;
import com.github.muehmar.gradle.openapi.generator.java.model.JavaPojoMembers;
import com.github.muehmar.gradle.openapi.generator.model.Necessity;
import com.github.muehmar.gradle.openapi.generator.model.Nullability;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Constraints;
import com.github.muehmar.gradle.openapi.generator.model.constraints.Pattern;
import com.github.muehmar.gradle.openapi.generator.settings.GetterSuffixes;
import com.github.muehmar.gradle.openapi.generator.settings.GetterSuffixesBuilder;
import com.github.muehmar.gradle.openapi.generator.settings.JavaModifier;
import com.github.muehmar.gradle.openapi.generator.settings.JsonSupport;
import com.github.muehmar.gradle.openapi.generator.settings.PojoSettings;
import com.github.muehmar.gradle.openapi.generator.settings.RawGetter;
import com.github.muehmar.gradle.openapi.generator.settings.TestPojoSettings;
import io.github.muehmar.codegenerator.Generator;
import io.github.muehmar.codegenerator.writer.Writer;
import org.junit.jupiter.api.Test;

class RequiredNullableGetterTest {

  @Test
  void generator_when_enabledJackson_then_correctOutputAndRefs() {
    final Generator<JavaPojoMember, PojoSettings> generator = RequiredNullableGetter.getter();
    final JavaPojoMember pojoMember =
        JavaPojoMembers.birthdate(Necessity.REQUIRED, Nullability.NULLABLE);

    final Writer writer =
        generator.generate(pojoMember, TestPojoSettings.defaultSettings(), Writer.createDefault());

    assertEquals(5, writer.getRefs().toHashSet().size());
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_TIME_LOCAL_DATE::equals));
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_UTIL_OPTIONAL::equals));
    assertTrue(writer.getRefs().exists(JacksonRefs.JSON_IGNORE::equals));
    assertTrue(writer.getRefs().exists(JacksonRefs.JSON_PROPERTY::equals));
    assertTrue(writer.getRefs().exists(JavaValidationRefs.ASSERT_TRUE::equals));
    assertEquals(
        "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "@JsonIgnore\n"
            + "public Optional<LocalDate> getBirthdate() {\n"
            + "  return Optional.ofNullable(birthdate);\n"
            + "}\n"
            + "\n"
            + "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "@JsonIgnore\n"
            + "public LocalDate getBirthdateOr(LocalDate defaultValue) {\n"
            + "  return birthdate == null ? defaultValue : birthdate;\n"
            + "}\n"
            + "\n"
            + "@JsonProperty(\"birthdate\")\n"
            + "private LocalDate getBirthdateRaw() {\n"
            + "  return birthdate;\n"
            + "}\n"
            + "\n"
            + "@AssertTrue(message = \"birthdate is required but it is not present\")\n"
            + "private boolean isBirthdatePresent() {\n"
            + "  return isBirthdatePresent;\n"
            + "}",
        writer.asString());
  }

  @Test
  void generator_when_disabledJackson_then_correctOutputAndRefs() {
    final Generator<JavaPojoMember, PojoSettings> generator = RequiredNullableGetter.getter();
    final JavaPojoMember pojoMember =
        JavaPojoMembers.birthdate(
            Constraints.ofPattern(Pattern.ofUnescapedString("DatePattern")),
            Necessity.REQUIRED,
            Nullability.NULLABLE);

    final Writer writer =
        generator.generate(
            pojoMember,
            TestPojoSettings.defaultSettings().withJsonSupport(JsonSupport.NONE),
            Writer.createDefault());

    assertEquals(4, writer.getRefs().toHashSet().size());
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_TIME_LOCAL_DATE::equals));
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_UTIL_OPTIONAL::equals));
    assertTrue(writer.getRefs().exists(JavaValidationRefs.ASSERT_TRUE::equals));
    assertTrue(writer.getRefs().exists(JavaValidationRefs.PATTERN::equals));
    assertEquals(
        "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "public Optional<LocalDate> getBirthdate() {\n"
            + "  return Optional.ofNullable(birthdate);\n"
            + "}\n"
            + "\n"
            + "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "public LocalDate getBirthdateOr(LocalDate defaultValue) {\n"
            + "  return birthdate == null ? defaultValue : birthdate;\n"
            + "}\n"
            + "\n"
            + "@Pattern(regexp=\"DatePattern\")\n"
            + "private LocalDate getBirthdateRaw() {\n"
            + "  return birthdate;\n"
            + "}\n"
            + "\n"
            + "@AssertTrue(message = \"birthdate is required but it is not present\")\n"
            + "private boolean isBirthdatePresent() {\n"
            + "  return isBirthdatePresent;\n"
            + "}",
        writer.asString());
  }

  @Test
  void generator_when_disabledJacksonAndValidation_then_correctOutputAndRefs() {
    final Generator<JavaPojoMember, PojoSettings> generator = RequiredNullableGetter.getter();
    final JavaPojoMember pojoMember =
        JavaPojoMembers.birthdate(Necessity.REQUIRED, Nullability.NULLABLE);

    final Writer writer =
        generator.generate(
            pojoMember,
            TestPojoSettings.defaultSettings()
                .withJsonSupport(JsonSupport.NONE)
                .withEnableConstraints(false),
            Writer.createDefault());

    assertEquals(2, writer.getRefs().toHashSet().size());
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_TIME_LOCAL_DATE::equals));
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_UTIL_OPTIONAL::equals));
    assertEquals(
        "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "public Optional<LocalDate> getBirthdate() {\n"
            + "  return Optional.ofNullable(birthdate);\n"
            + "}\n"
            + "\n"
            + "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "public LocalDate getBirthdateOr(LocalDate defaultValue) {\n"
            + "  return birthdate == null ? defaultValue : birthdate;\n"
            + "}",
        writer.asString());
  }

  @Test
  void generator_when_requiredNullableSuffix_then_correctOutputAndRefs() {
    final Generator<JavaPojoMember, PojoSettings> generator = RequiredNullableGetter.getter();
    final JavaPojoMember pojoMember =
        JavaPojoMembers.birthdate(Necessity.REQUIRED, Nullability.NULLABLE);

    final GetterSuffixes getterSuffixes =
        GetterSuffixesBuilder.create()
            .requiredSuffix("")
            .requiredNullableSuffix("ReqNull")
            .optionalSuffix("")
            .optionalNullableSuffix("")
            .build();

    final Writer writer =
        generator.generate(
            pojoMember,
            TestPojoSettings.defaultSettings()
                .withJsonSupport(JsonSupport.NONE)
                .withEnableConstraints(false)
                .withGetterSuffixes(getterSuffixes),
            Writer.createDefault());

    assertEquals(2, writer.getRefs().toHashSet().size());
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_TIME_LOCAL_DATE::equals));
    assertTrue(writer.getRefs().exists(JavaRefs.JAVA_UTIL_OPTIONAL::equals));
    assertEquals(
        "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "public Optional<LocalDate> getBirthdateReqNull() {\n"
            + "  return Optional.ofNullable(birthdate);\n"
            + "}\n"
            + "\n"
            + "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "public LocalDate getBirthdateOr(LocalDate defaultValue) {\n"
            + "  return birthdate == null ? defaultValue : birthdate;\n"
            + "}",
        writer.asString());
  }

  @Test
  void generator_when_deprecatedAnnotation_then_correctOutputAndRefs() {
    final Generator<JavaPojoMember, PojoSettings> generator = RequiredNullableGetter.getter();
    final JavaPojoMember pojoMember =
        JavaPojoMembers.birthdate(Necessity.REQUIRED, Nullability.NULLABLE);

    final RawGetter rawGetter =
        TestPojoSettings.defaultRawGetter()
            .withDeprecatedAnnotation(true)
            .withModifier(JavaModifier.PUBLIC);

    final Writer writer =
        generator.generate(
            pojoMember,
            TestPojoSettings.defaultSettings()
                .withJsonSupport(JsonSupport.JACKSON)
                .withRawGetter(rawGetter),
            Writer.createDefault());

    assertEquals(
        "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "@JsonIgnore\n"
            + "public Optional<LocalDate> getBirthdate() {\n"
            + "  return Optional.ofNullable(birthdate);\n"
            + "}\n"
            + "\n"
            + "/**\n"
            + " * Birthdate\n"
            + " */\n"
            + "@JsonIgnore\n"
            + "public LocalDate getBirthdateOr(LocalDate defaultValue) {\n"
            + "  return birthdate == null ? defaultValue : birthdate;\n"
            + "}\n"
            + "\n"
            + "@JsonProperty(\"birthdate\")\n"
            + "@Deprecated\n"
            + "public LocalDate getBirthdateRaw() {\n"
            + "  return birthdate;\n"
            + "}\n"
            + "\n"
            + "@AssertTrue(message = \"birthdate is required but it is not present\")\n"
            + "@Deprecated\n"
            + "public boolean isBirthdatePresent() {\n"
            + "  return isBirthdatePresent;\n"
            + "}",
        writer.asString());
  }
}
