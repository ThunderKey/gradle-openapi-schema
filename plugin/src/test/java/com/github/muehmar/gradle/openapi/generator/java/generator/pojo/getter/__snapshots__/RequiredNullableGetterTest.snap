com.github.muehmar.gradle.openapi.generator.java.generator.pojo.getter.RequiredNullableGetterTest.generator_when_deprecatedAnnotation_then_correctOutputAndRefs=[
com.fasterxml.jackson.annotation.JsonIgnore
com.fasterxml.jackson.annotation.JsonProperty
java.time.LocalDate
java.util.Optional
javax.validation.constraints.AssertTrue

/**
 * Birthdate
 */
@JsonIgnore
public Optional<LocalDate> getBirthdate() {
  return Optional.ofNullable(birthdate);
}

/**
 * Birthdate
 */
@JsonIgnore
public LocalDate getBirthdateOr(LocalDate defaultValue) {
  return this.birthdate == null ? defaultValue : this.birthdate;
}

/**
 * @deprecated This method is intended to be used only by frameworks for validation
 * or serialisation.
 */
@JsonProperty("birthdate")
@Deprecated
public LocalDate getBirthdateRaw() {
  return birthdate;
}

/**
 * @deprecated This method is intended to be used only by frameworks for validation
 * or serialisation.
 */
@AssertTrue(message = "birthdate is required but it is not present")
@Deprecated
public boolean isBirthdatePresent() {
  return isBirthdatePresent;
}
]


com.github.muehmar.gradle.openapi.generator.java.generator.pojo.getter.RequiredNullableGetterTest.generator_when_disabledJacksonAndValidation_then_correctOutputAndRefs=[
java.time.LocalDate
java.util.Optional

/**
 * Birthdate
 */
public Optional<LocalDate> getBirthdate() {
  return Optional.ofNullable(birthdate);
}

/**
 * Birthdate
 */
public LocalDate getBirthdateOr(LocalDate defaultValue) {
  return this.birthdate == null ? defaultValue : this.birthdate;
}
]


com.github.muehmar.gradle.openapi.generator.java.generator.pojo.getter.RequiredNullableGetterTest.generator_when_disabledJackson_then_correctOutputAndRefs=[
java.time.LocalDate
java.util.Optional
javax.validation.constraints.AssertTrue
javax.validation.constraints.Pattern

/**
 * Birthdate
 */
public Optional<LocalDate> getBirthdate() {
  return Optional.ofNullable(birthdate);
}

/**
 * Birthdate
 */
public LocalDate getBirthdateOr(LocalDate defaultValue) {
  return this.birthdate == null ? defaultValue : this.birthdate;
}

@Pattern(regexp="DatePattern")
private LocalDate getBirthdateRaw() {
  return birthdate;
}

@AssertTrue(message = "birthdate is required but it is not present")
private boolean isBirthdatePresent() {
  return isBirthdatePresent;
}
]


com.github.muehmar.gradle.openapi.generator.java.generator.pojo.getter.RequiredNullableGetterTest.generator_when_enabledJackson_then_correctOutputAndRefs=[
com.fasterxml.jackson.annotation.JsonIgnore
com.fasterxml.jackson.annotation.JsonProperty
java.time.LocalDate
java.util.Optional
javax.validation.constraints.AssertTrue

/**
 * Birthdate
 */
@JsonIgnore
public Optional<LocalDate> getBirthdate() {
  return Optional.ofNullable(birthdate);
}

/**
 * Birthdate
 */
@JsonIgnore
public LocalDate getBirthdateOr(LocalDate defaultValue) {
  return this.birthdate == null ? defaultValue : this.birthdate;
}

@JsonProperty("birthdate")
private LocalDate getBirthdateRaw() {
  return birthdate;
}

@AssertTrue(message = "birthdate is required but it is not present")
private boolean isBirthdatePresent() {
  return isBirthdatePresent;
}
]


com.github.muehmar.gradle.openapi.generator.java.generator.pojo.getter.RequiredNullableGetterTest.generator_when_requiredNullableSuffix_then_correctOutputAndRefs=[
java.time.LocalDate
java.util.Optional

/**
 * Birthdate
 */
public Optional<LocalDate> getBirthdateReqNull() {
  return Optional.ofNullable(birthdate);
}

/**
 * Birthdate
 */
public LocalDate getBirthdateOr(LocalDate defaultValue) {
  return this.birthdate == null ? defaultValue : this.birthdate;
}
]